src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -150,42 +279,45 @@ const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelec
         selectedRef = selectedItem;
       }
 
+      visibleNodes.push({ id, parent });
       const menuItemClassNames = classNames(
         cx([
           'item',
           { 'is-selected': isSelected },
         ]),
       );
 
+      const optionalAttributes = itemHasChildren ? { 'aria-expanded': itemIsOpen, 'aria-haspopup': true } : {};
+
       return (
         <React.Fragment key={item.path}>
           <div className={!firstLevel ? cx('indent') : null}>
```

Comment:
It'd be best to move this render logic into its own subcomponent, as is it makes this file quite large and messy to navigate.

Reply:
Would I be able to nest this subcomponent within the navigation menu component? At the very least, I would need to pass the `openKeys` prop to the new subcomponent, so keeping it within scope would be convenient.

Reply:
@dkasper-was-taken  , in this case do you think it makes sense to make a new file for the menuItem subcomponent? the _CollapsingNavigationMenu.jsx file is already over 300 lines long, so it might be nice to shorten it a bit.

Reply:
Yes, a separate file would be ideal. Outside of some documentation examples it's our common practice for each component and sub-component to have their own file.

Reply:
Just finished extracting the menu items to their own file.